components/eamxx/src/physics/nudging/eamxx_nudging_process_interface.cpp

```diff
@@ -91,6 +101,30 @@ void Nudging::apply_tendency(Field& base, const Field& next, const int dt)
   base.update(tend,dtend,Real(1.0));
 }
 // =========================================================================================
+void Nudging::apply_weighted_tendency(Field& base, const Field& next, const Field& weights, const int dt)
+{
+  // Calculate the weight to apply the tendency
+  const Real dtend = Real(dt)/Real(m_timescale);
```

Comment:
is dt the model timestep? I'm surprised it has integer type.

Reply:
Yes, the model timestep is passed as an integer to atmosphere processes.  I can't quite remember why, @bartgol may.

Reply:
I think E3SM/CPL7 may be specifying dt as an integer (which saves a bit of memory), so it would be natural for us to do so as well. Still, I'm surprised we don't recast it to double. In particular, the ability to substep procs at runtime seems like it would naturally result in non-integer timesteps. For example, what if full-model dt is 150 sec and a user asks for 7 micro substeps (=21.43 sec micro dt)?

Reply:
The timestep is _not_ an integer. It is passed by the cpl to the AD as an integer, but atm processes expect a double.

Reply:
Ok, so this should be fixed.

Reply:
@xyuan we still need to fix dt here. The method should accept a Real, not an integer, to avoid silent truncation by the compiler.